Helper/Webhook/OrderClosedWebhookHandler.php

```diff
@@ -71,7 +79,50 @@ public function handleWebhook(
         Notification $notification,
         string $transitionState
     ): MagentoOrder {
+        $additionalData = $notification->getAdditionalData();
+        if (!empty($additionalData)) {
+            $additionalData = $this->serializer->unserialize($additionalData);
+        }
+
         if ($notification->isSuccessful()) {
+            foreach ($additionalData as $key => $value) {
+                // Check if the key matches the pattern "order-X-pspReference"
+                if (preg_match('/^order-(\d+)-pspReference$/', $key, $matches)) {
+                    $orderIndex = (int)$matches[1];
+                    $pspReference = $value;
+                    $sortValue = $orderIndex;
+
+                    // Retrieve adyen_order_payment for this pspReference
+                    $adyenOrderPayment = $this->adyenOrderPaymentCollectionFactory->create()
+                        ->addFieldToFilter('pspreference', $pspReference)
```

Comment:
As `pspreference` field is being used as a filter, shall we add an index for this field to the DB definition?

Reply:
Ass discussed, we already have an UNIQUE index defined for pspreference and that would work.